powerline/segments/i3wm.py

```diff
@@ -115,3 +115,37 @@ def mode(pl, segment_info, names={'default': None}):
 	if mode in names:
 		return names[mode]
 	return mode
+
+
+def scratchpad_groups(w):
+	group = []
+	if w.urgent:
+		group.append('s_urgent')
+	if w.nodes[0].focused:
+		group.append('s_focused')
+	if w.workspace().name != '__i3_scratch':
+		group.append('s_visible')
```

Comment:
Group must be named `scratchpad:visible`, same for other. `w_*` are also invalid names, but they cannot be changed for backwards compatibility #1520.